bundler/spec/bundler/lockfile_parser_spec.rb

```diff
@@ -95,6 +95,154 @@
     end
   end
 
+  describe "X64_MINGW_LEGACY platform handling" do
+    before { allow(Bundler::SharedHelpers).to receive(:find_gemfile).and_return(bundled_app("gems.rb")) }
+
+    describe "when X64_MINGW_LEGACY is present alone" do
+      let(:lockfile_with_legacy_platform) { <<~L }
+        GIT
+          remote: https://github.com/alloy/peiji-san.git
+          revision: eca485d8dc95f12aaec1a434b49d295c7e91844b
+          specs:
+            peiji-san (1.2.0)
+
+        GEM
+          remote: https://rubygems.org/
+          specs:
+            rake (10.3.2)
+
+        PLATFORMS
+          ruby
+          x64-mingw32
+
+        DEPENDENCIES
+          peiji-san!
+          rake
+
+        BUNDLED WITH
+           1.12.0.rc.2
+      L
+
+      context "when bundle is not frozen" do
+        before { allow(Bundler).to receive(:frozen_bundle?).and_return(false) }
+        subject { described_class.new(lockfile_with_legacy_platform) }
+
+        it "replaces X64_MINGW_LEGACY with X64_MINGW" do
+          allow(Bundler::SharedHelpers).to receive(:major_deprecation)
+          expect(subject.platforms.map(&:to_s)).to contain_exactly("ruby", "x64-mingw-ucrt")
+          expect(subject.platforms.map(&:to_s)).not_to include("x64-mingw32")
+        end
+
+        it "shows deprecation warning for replacement" do
+          expect(Bundler::SharedHelpers).to receive(:major_deprecation).with(
+            2,
+            "Found x64-mingw32 in lockfile, which is deprecated. Using the current x64-mingw-ucrt platform instead. Support for x64-mingw32 will be removed in Bundler 3.0.",
+            removed_message: "Found x64-mingw32 in lockfile, which is no longer supported as of Bundler 3.0. Using the current x64-mingw-ucrt platform instead."
+          )
+          subject
+        end
+      end
+
+      context "when bundle is frozen" do
+        before { allow(Bundler).to receive(:frozen_bundle?).and_return(true) }
+        subject { described_class.new(lockfile_with_legacy_platform) }
+
+        it "preserves X64_MINGW_LEGACY platform without replacement" do
+          expect(subject.platforms.map(&:to_s)).to contain_exactly("ruby", "x64-mingw32")
+        end
+
+        it "does not show any deprecation warnings" do
+          expect(Bundler::SharedHelpers).not_to receive(:major_deprecation)
+          subject
+        end
+      end
+    end
+
+    describe "when both X64_MINGW_LEGACY and X64_MINGW are present" do
+      let(:lockfile_with_both_platforms) { <<~L }
+        GIT
+          remote: https://github.com/alloy/peiji-san.git
+          revision: eca485d8dc95f12aaec1a434b49d295c7e91844b
+          specs:
+            peiji-san (1.2.0)
+
```

Comment:
Since these specs are unrelated to git sources, should we remove this section and use a simpler lockfile?

Reply:
Cleaned them up to a pretty minimal size/shape.